### PR TITLE
HDDS-9276. Mark disabled tests using custom tag/category.

### PR DESCRIPTION
**Pull request by devmadhuu**
## What changes were proposed in this pull request?

The goal of this PR is to replace these `@Ignore` and `@Disabled` annotations with conditional ones. Regular CI should still skip them, but we would like to include them in nightly or other scheduled runs.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9276

## How was this patch tested?

This patch is tested with existing CI workflow run of unit and integration tests where tests annotated with "Unhealthy" annotation would not run.

### CODE REVIEW

File: hadoop-ozone/dev-support/checks/unit.sh [hunk 1]
```diff
@@ -16,4 +16,4 @@
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 CHECK=unit
-source "${DIR}/junit.sh" -pl \!:ozone-integration-test,\!:mini-chaos-tests -DexcludedGroups=native "$@"
+source "${DIR}/junit.sh" -pl \!:ozone-integration-test,\!:mini-chaos-tests -DexcludedGroups=native | unhealthy "$@"
```
Comment (adoroszlai) [inline hadoop-ozone/dev-support/checks/unit.sh:19]: Need to quote the parameter value, as `|` is a special symbol for the shell, too.  And the category is needed for JUnit4.

```suggestion
source "${DIR}/junit.sh" -pl \!:ozone-integration-test,\!:mini-chaos-tests -DexcludedGroups="native | unhealthy | org.apache.ozone.test.UnhealthyTest" "$@"
```

Comment (devmadhuu) [reply hadoop-ozone/dev-support/checks/unit.sh]: Thanks @adoroszlai for proactively reviewing it. Handled the above issue.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileInterfacesWithFSO.java [hunk 2]
```diff
@@ -43,35 +45,35 @@ public TestOzoneFileInterfacesWithFSO(boolean setDefaultFs,
 
   @Override
   @Test
-  @Ignore("HDDS-2939")
+  @Category(UnhealthyTest.class) @Unhealthy("HDDS-2939")
   public void testReplication() throws IOException {
     // ignore as this is not relevant to PREFIX layout changes
```
[the remark this answers is not in the document]
Comment (devmadhuu) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileInterfacesWithFSO.java]: Ok @adoroszlai. I have better kept them ignored as earlier.

[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileInterfacesWithFSO.java]: Thanks, but the same applies to two other FSO test classes, too:

hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystemWithFSO.java
hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedOzoneFileSystemWithFSO.java

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystemWithFSO.java [hunk 3]
```diff
@@ -85,6 +85,12 @@ public TestOzoneFileSystemWithFSO(boolean setDefaultFs,
   @Override
   public void cleanup() {
     super.cleanup();
+    try {
+      deleteRootDir();
+    } catch (IOException e) {
+      LOG.info("Failed to cleanup DB tables.", e);
+      fail("Failed to cleanup DB tables." + e.getMessage());
+    }
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystemWithFSO.java:93]: This recently added part seems unrelated.

**Review by adoroszlai — COMMENTED** [review]
Thanks @devmadhuu for updating the patch.  Looks mostly good.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystemWithFSO.java (hunk 3 above)
Comment (devmadhuu) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystemWithFSO.java]: > This recently added part seems unrelated.

Ok, its handled. I think it came while resolving conflict. Checked carefully.

**Review by adoroszlai — APPROVED** [review]
Thanks @devmadhuu for updating the patch, latest version LGTM.
